koku/api/report/aws/query_handler.py

```diff
@@ -523,13 +523,19 @@ def execute_individual_query(self, org_unit_applied=False):  # noqa: C901
                     tag_results = self._get_associated_tags(query_table, self.query_filter)
 
             query_sum = self._build_sum(query, annotations)
-
-            if self._limit and not org_unit_applied:
-                rank_order = getattr(F(self.order_field), self.order_direction)()
-                rank_by_total = Window(expression=RowNumber(), partition_by=F("date"), order_by=rank_order)
-                query_data = query_data.annotate(rank=rank_by_total)
-                query_order_by.insert(1, "rank")
-                query_data = self._ranked_list(query_data)
+            # Set default delta if one was not passed in.
+
+            if not org_unit_applied:
+                if self._limit and query_data:
+                    rank_orders = []
+                    if self.order_field == "delta":
+                        rank_orders.append(getattr(F(self._delta), self.order_direction)())
+                    else:
+                        rank_orders.append(getattr(F(self.order_field), self.order_direction)())
+                    rank_by_total = Window(expression=RowNumber(), partition_by=F("date"), order_by=rank_orders)
+                    query_data = query_data.annotate(rank=rank_by_total)
+                    query_order_by.insert(1, "rank")
+                    query_data = self._ranked_list(query_data)
```

Comment:
I don't think you need the extra level of indentation

```suggestion
            if self._limit and query_data and not org_unit_applied:
                rank_orders = []
                if self.order_field == "delta":
                    rank_orders.append(getattr(F(self._delta), self.order_direction)())
                else:
                    rank_orders.append(getattr(F(self.order_field), self.order_direction)())
                rank_by_total = Window(expression=RowNumber(), partition_by=F("date"), order_by=rank_orders)
                query_data = query_data.annotate(rank=rank_by_total)
                query_order_by.insert(1, "rank")
                query_data = self._ranked_list(query_data)
```